components/eamxx/scripts/atmchange

```diff
@@ -52,6 +52,12 @@ def atm_config_chg(changes, reset=False, all_matches=False, buffer_only=False):
     else:
         expect(changes, "Missing <param>=<val> args")
 
+    # Before applying/buffering changes, at the very least check the syntax
+    for c in changes:
+        # This will throw if the syntax is bad
+        _, _, _ = parse_change(c)
+
+    any_change = True
```

Comment:
I think `any_change` should be initialized to `False`. Since we are OR-ing them together, initializing to `True` will mean it will always be `True`.

Reply:
It is inited to False inside the `if not buffer_only` case. But if `buffer_only=True`, we must assume there are changes, since we have no way to check it (unless we go through them all). It is a bit redundant. I can remove one line and init it as `any_change = buffer_only`.